src/sentry/notifications/platform/providers/base.py

```diff
@@ -0,0 +1,44 @@
+import abc
+from enum import StrEnum
+from typing import Generic, TypeVar
+
+from sentry.organizations.services.organization.model import RpcOrganizationSummary
+
+ProviderRenderable = TypeVar("ProviderRenderable")
+"""
+A renderable object that is understood by the notification provider.
+For example, Email might expect HTML, or raw text; Slack might expect a JSON Block Kit object.
+"""
+
+
+class NotificationProviderKey(StrEnum):
+    """
+    The unique keys for each registered notification provider.
+    """
+
+    EMAIL = "email"
+    SLACK = "slack"
+    MSTEAMS = "msteams"
+    DISCORD = "discord"
+
```

Comment:
We should be able to derive most of these from the shared `IntegrationProviderSlug` enum, just to make sure we keep these in sync: https://github.com/getsentry/sentry/blob/f28055b2373a75a348b8e213880422952fbb5cc3/src/sentry/integrations/types.py#L31